library/sources/FastXmlParser.ts

```diff
@@ -38,18 +38,39 @@ export class FastXmlParser implements Wrapper {
   }
 
   wrap(hooks: Hooks) {
-    hooks
-      .addPackage("fast-xml-parser")
-      .withVersion("^4.0.0")
-      .onRequire((exports, pkgInfo) => {
-        wrapNewInstance(exports, "XMLParser", pkgInfo, (instance) => {
+    const pkg = hooks.addPackage("fast-xml-parser");
+    pkg.withVersion("^4.0.0").onRequire((exports, pkgInfo) => {
+      wrapNewInstance(exports, "XMLParser", pkgInfo, (instance) => {
+        wrapExport(instance, "parse", pkgInfo, {
+          modifyReturnValue: (args, returnValue) => {
+            this.inspectParse(args, returnValue);
+            return returnValue;
+          },
+        });
+      });
+    });
+
+    pkg.withVersion("^5.0.0").onRequire((exports, pkgInfo) => {
+      const parser = exports.XMLParser; // It's a getter, so we can't directly pass it to wrapNewInstance
+
+      const wrappedParser = wrapNewInstance(
+        parser,
+        undefined,
+        pkgInfo,
+        (instance) => {
           wrapExport(instance, "parse", pkgInfo, {
             modifyReturnValue: (args, returnValue) => {
               this.inspectParse(args, returnValue);
               return returnValue;
             },
           });
-        });
-      });
+        }
+      );
+
+      return {
+        ...exports,
+        XMLParser: wrappedParser,
+      };
```

Comment:
would the approach for v5 also work for v4?

Reply:
Why didn't I think of that? 😄